From 483d35b949685f473e9ce8e240d64c086efc04d9 Mon Sep 17 00:00:00 2001
From: Vladislav Buzov <vbuzov@embeddedalley.com>
Date: Tue, 9 Jun 2009 16:01:15 -0500
Subject: [PATCH] [Bug 77] ALSA sample rate is off for 22050Hz and 32KHz

Added a constraint for ALSA library to create buffers multiple
of period size.
---
 sound/soc/stmp3xxx/stmp3xxx_pcm.c |    6 ++++++
 1 files changed, 6 insertions(+), 0 deletions(-)

diff --git a/sound/soc/stmp3xxx/stmp3xxx_pcm.c b/sound/soc/stmp3xxx/stmp3xxx_pcm.c
index 79fd552..bbd6d75 100644
--- a/sound/soc/stmp3xxx/stmp3xxx_pcm.c
+++ b/sound/soc/stmp3xxx/stmp3xxx_pcm.c
@@ -324,6 +324,12 @@ static int stmp3xxx_pcm_open(struct snd_pcm_substream *substream)
 	struct stmp3xxx_runtime_data *prtd;
 	int ret;
 
+	/* Ensure that buffer size is a multiple of the period size */
+	ret = snd_pcm_hw_constraint_integer(runtime,
+					SNDRV_PCM_HW_PARAM_PERIODS);
+	if (ret < 0)
+		return ret;
+
 	snd_soc_set_runtime_hwparams(substream, &stmp3xxx_pcm_hardware);
 
 	prtd = kzalloc(sizeof(struct stmp3xxx_runtime_data), GFP_KERNEL);
-- 
1.5.4.4

